hypervisor/arch/x86/vmx.c

```diff
@@ -481,13 +506,24 @@ static bool is_cr4_write_valid(uint64_t cr4)
 void vmx_write_cr4(struct vcpu *vcpu, uint64_t cr4)
 {
 	uint64_t cr4_vmx;
+	uint64_t old_cr4 = vcpu_get_cr4(vcpu);
 
-	if (!is_cr4_write_valid(cr4)) {
+	if (!is_cr4_write_valid(vcpu, cr4)) {
 		pr_dbg("Invalid cr4 write operation from guest");
 		vcpu_inject_gp(vcpu, 0U);
 		return;
 	}
 
+	if ((cr4 ^ old_cr4) & (CR4_PGE | CR4_PSE | CR4_PAE |
```

Comment:
for conditions, please use " a == b" or "a!= b" etc.. Please fix in all of the files.

Reply:
done